app/models/exercise.rb

```diff
@@ -6,17 +6,41 @@ def self.create_exercise(blueprint_id, workout_id, sets, user_id)
     @blueprint = Blueprint.find(blueprint_id)
     exercises = @blueprint.exercises.where(workout_id: nil)
     exercises.each do |exercise|
+      weight, reps =
+        exercise.set_weight_and_reps(user_id).values_at(:weight, :reps)
+      #  debugger
       new_exercise =
         Exercise.create!(
           name: exercise["name"],
           instructions: exercise["instructions"],
           blueprint_id: exercise["blueprint_id"],
           pic_url: exercise["pic_url"],
-          workout_id: workout_id
+          workout_id: workout_id,
+          weight: weight,
+          reps: reps
         )
       sets.times do
         ExerciseSet.create_exercise_sets new_exercise["id"], user_id
       end
     end
   end
+
+  def set_weight_and_reps(user_id)
```

Comment:
Since this method has a lot of logic in it, probably worth putting a sentence or two about what the output of this method is.